src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrBackflower.cs

```diff
@@ -0,0 +1,323 @@
+        var submoduleExclusions = _sourceManifest.Submodules
+            .Where(s => s.Path.StartsWith(mapping.Name + '/'))
+            .Select(s => s.Path.Substring(mapping.Name.Length + 1))
+            .Select(VmrPatchHandler.GetExclusionRule)
+            .ToList();
+
+        // TODO: Remove VMR patches before we create the patches
+
+        List<VmrIngestionPatch> patches = await _vmrPatchHandler.CreatePatches(
+            patchName,
+            Constants.EmptyGitObject,
+            shaToFlow,
+            path: null,
+            submoduleExclusions,
+            relativePaths: true,
+            workingDir: _vmrInfo.GetRepoSourcesPath(mapping),
+            applicationPath: null,
+            cancellationToken);
+
+        _logger.LogInformation("Created {count} patch(es)", patches.Count);
+
+        // We will remove everything not-cloaked and replace it with current contents of the source repo
+        // When flowing to a repo, we remove all repo files but submodules and cloaked files
+        List<string> removalFilters =
+        [
+            .. mapping.Include.Select(VmrPatchHandler.GetInclusionRule),
+            .. mapping.Exclude.Select(VmrPatchHandler.GetExclusionRule),
+            .. submoduleExclusions,
+        ];
+
+        ProcessExecutionResult result = await _processManager.ExecuteGit(targetRepo, ["rm", "-r", "-q", "--", .. removalFilters]);
+        result.ThrowIfFailed($"Failed to remove files from {targetRepo}");
+
+        // Now we insert the VMR files
+        foreach (var patch in patches)
+        {
+            // TODO: Handle exceptions
+            await _vmrPatchHandler.ApplyPatch(patch, targetRepo, cancellationToken);
+
+            if (discardPatches)
+            {
+                _fileSystem.DeleteFile(patch.Path);
+            }
+        }
+
+        // TODO: Check if there are any changes and only commit if there are
+        result = await _processManager.ExecuteGit(
+            targetRepo,
+            ["diff-index", "--quiet", "--cached", "HEAD", "--"],
+            cancellationToken: cancellationToken);
+
+        if (result.ExitCode == 0)
+        {
+            // TODO: Handle + clean up the work branch
+            return null;
+        }
+
+        var commitMessage = $"""
+            [VMR] Codeflow {shortShas}
+
+            {Constants.AUTOMATION_COMMIT_TAG}
+            """;
+
+        await _localGitClient.CommitAsync(targetRepo, commitMessage, false, cancellationToken: cancellationToken);
+        await _localGitClient.ResetWorkingTree(targetRepo);
+
+        return branchName;
+    }
+
+    private async Task UpdateVersionDetailsXml(NativePath repoPath, string currentVmrSha, CancellationToken cancellationToken)
+    {
+        // TODO: Do a proper full update of all dependencies, not just V.D.xml
+        var versionDetailsXml = DependencyFileManager.GetXmlDocument(await _fileSystem.ReadAllTextAsync(repoPath / VersionFiles.VersionDetailsXml));
```

Comment:
Yes, this is correct. It's expected that this will be re-done once we have a full list of dependencies. For that, we have this functionality already encapsulated in `DependencyFileManager` and no XMLs will be handled here.
There is a TODO above for that but we can't implement that yet as we don't have VMR BAR builds yet.